### PR TITLE
Fix #2355: shifted helper function to central utility testing module

### LINKED ISSUE #2355: Shift helper functions `hasErrorText` and `hasNoErrorText` to central utility testing file
**Is your feature request related to a problem? Please describe.**
Helper functions `hasErrorText` and `hasNoErrorText` is being used frequently in test files like `AdminAuthActivityTest` and it will be used in more files in future. 

**Describe the solution you'd like**
We should shift these to a central file rather than including them every time in the test files.

### PR DESCRIPTION
**Pull request by Akshat-gour**
## Explanation
This PR Fixes #2355: In which we Shifted helper function **hasErrorText** and **hasNoErrorText**  from **org.oppia.android.app(test)** > **profile** > ( **AddProfileActivityTest.kt** ,  **AdminAuthActivityTest.kt**  ,  **AdminPinActivityTest.kt**  ,  **PinPasswordActivityTest.kt** ) 
To **org.oppia.android.testing** > **EditTextInputAction.kt**  and added Context Parameter to EditTextInputAction Constructor and added Material Dependency to build.gradle(:testing) and to build.bazel(testing)

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

File: .idea/misc.xml [hunk 1]
```diff
@@ -1,9 +1,9 @@
...
 <project version="4">
-  <component name="ProjectRootManager" version="2" languageLevel="JDK_1_8" project-jdk-name="1.8" project-jdk-type="JavaSDK">
+  <component name="ProjectRootManager" version="2" languageLevel="JDK_1_8" default="true" project-jdk-name="1.8" project-jdk-type="JavaSDK">
     <output url="file://$PROJECT_DIR$/build/classes" />
   </component>
   <component name="ProjectType">
     <option name="id" value="Android" />
   </component>
-</project>
+</project>
```
Comment (prayutsu) [inline .idea/misc.xml:9]: Revert the changes in these files, these are auto-generated and we don't need them.

File: app/src/sharedTest/java/org/oppia/android/app/profile/AddProfileActivityTest.kt [hunk 2]
```diff
@@ -1529,30 +1529,7 @@ class AddProfileActivityTest {
     }
   }
 
-  private fun hasErrorText(@StringRes expectedErrorTextId: Int): Matcher<View> {
-    return object : TypeSafeMatcher<View>() {
-      override fun matchesSafely(view: View): Boolean {
-        val expectedErrorText = context.resources.getString(expectedErrorTextId)
-        return (view as TextInputLayout).error == expectedErrorText
-      }
```
Comment (prayutsu) [inline app/src/sharedTest/java/org/oppia/android/app/profile/AddProfileActivityTest.kt:1532]: Remove this blank line.
Ditto elsewhere.

File: testing/src/main/java/org/oppia/android/testing/EditTextInputAction.kt [hunk 3]
```diff
@@ -17,7 +22,7 @@ import javax.inject.Inject
  * for specifics.
  */
 class EditTextInputAction @Inject constructor(
-  val testCoroutineDispatchers: TestCoroutineDispatchers
+  val testCoroutineDispatchers: TestCoroutineDispatchers , val context: Context
 ) {
```
Comment (prayutsu) [inline testing/src/main/java/org/oppia/android/testing/EditTextInputAction.kt:26]: Add this context dependency in new line.

File: testing/src/main/java/org/oppia/android/testing/EditTextInputAction.kt [hunk 4]
```diff
@@ -43,4 +48,27 @@ class EditTextInputAction @Inject constructor(
       }
     }
   }
+  fun hasErrorText(@StringRes expectedErrorTextId: Int): Matcher<View> {
```
Comment (prayutsu) [inline testing/src/main/java/org/oppia/android/testing/EditTextInputAction.kt:51]: Add an empty line before the function, ditto below.

**Review by prayutsu — CHANGES_REQUESTED** [review]
@Akshat-gour Took a quick pass, mostly requested nit changes. PTAL.

File: .idea/misc.xml (hunk 1 above)
Comment (Akshat-gour) [reply .idea/misc.xml:9]: @prayutsu 
should i delete this file
